src/fmt.rs

```diff
@@ -0,0 +1,23 @@
+use termcolor::{Color, ColorChoice, ColorSpec, StandardStream, WriteColor};
+
+pub fn bold<F>(write: F)
+where
+    F: FnOnce(&mut StandardStream) -> std::io::Result<()>,
+{
+    let mut stdout = StandardStream::stdout(ColorChoice::Auto);
+    stdout.set_color(ColorSpec::new().set_bold(true)).ok();
+    write(&mut stdout).expect("Unexpected error writing to stdout");
+    stdout.reset().ok();
+}
+
+pub fn with_color_maybe_bold<F>(write: F, color: Color, bold: bool)
+where
+    F: FnOnce(&mut StandardStream) -> std::io::Result<()>,
+{
+    let mut stdout = StandardStream::stdout(ColorChoice::Auto);
+    stdout
+        .set_color(ColorSpec::new().set_fg(Some(color)).set_bold(bold))
+        .ok();
+    write(&mut stdout).expect("Unexpected error writing to stdout");
+    stdout.reset().ok();
+}
```

Comment:
Rather than using `bool` literals everywhere for `bold`, we should add dedicated functions for each behaviour, e.g. `colored` and `bold_colored`.

Also, in Rust, it's generally more conventional to have values precede functions within function arguments, e.g.

```rust
fn colored_bold<F>(color: Color, write: F)
```

This results in nicer formatting in the case that a large multi-line closure is provided, e.g.

```rust
colored_bold(Color::Green, |s| {
    writeln!(s, "blah blah blah: {}", foo);
    // more lines...
});
```
rather than
```rust
colored_bold(
    |s| {
        writeln!(s, "blah blah blah: {}", foo);
        // more lines...
    },
    Color::Green,
);

Reply:
Fixed with comment in [ad91ea4](https://github.com/FuelLabs/fuelup/pull/64/commits/ad91ea43ef7b20cdba817090bfb9ec3745f89f09) - is the comment clear?